extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -367,6 +376,11 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     this.troubleshootData_.slotId = this.element.getAttribute('data-slot');
     this.troubleshootData_.slotIndex =
         this.element.getAttribute('data-amp-slot-index');
+    if (!this.isFluidRequest_) {
+      const multiSizeStr = this.element.getAttribute('data-multi-size');
+      this.isFluidRequest_ = !!multiSizeStr &&
```

Comment:
I'm confused how this interacts with setting `isFluidRequest_` in `isLayoutSupported`.  Does the AMP runtime guarantee that `isLayoutSupported` will be called?  And is it called before `buildCallback`?

Reply:
Yes, `isLayoutSupported` is always called, and before `buildCallback`. Unfortunately, I don't have any docs to point to, but you can trace the code from here: https://github.com/ampproject/amphtml/blob/master/src/custom-element.js#L411. The idea is that `isFluidPrimaryRequest_` always implies `isFluidRequest_`, so if `isFluidPrimaryRequest_` was set in `isLayoutSupported`, then `isFluidRequest_` would also be set, and we wouldn't need to check this condition here.